test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,586 @@
+    {
+      baseInputCloud->emplace_back(0.0f, 0.0f, 0.0f);
+      baseInputCloud->emplace_back(0.0f, 0.0f, 1.0f);
+      baseInputCloud->emplace_back(0.0f, 1.0f, 0.0f);
+      baseInputCloud->emplace_back(0.0f, 1.0f, 1.0f);
+      baseInputCloud->emplace_back(1.0f, 0.0f, 0.0f);
+      baseInputCloud->emplace_back(1.0f, 0.0f, 1.0f);
+      baseInputCloud->emplace_back(1.0f, 1.0f, 0.0f);
+      baseInputCloud->emplace_back(1.0f, 1.0f, 1.0f);
+    }
+    protected:
+
+      void test(std::function<void(pcl::CropHull<pcl::PointXYZ>, TestData const &)> checker) {
+        pcl::PointCloud<pcl::PointXYZ>::Ptr inputCloud (new pcl::PointCloud<pcl::PointXYZ>);
+        for (pcl::PointXYZ const & baseOffset : baseOffsetList)
+        {
+          pcl::copyPointCloud(*baseInputCloud, *inputCloud);
+          for (pcl::PointXYZ & p : *inputCloud) {
+            p.getVector3fMap() += baseOffset.getVector3fMap();
+          }
+          auto insidePointGenerator = [this, &baseOffset] () {
+            pcl::PointXYZ p(rd(gen), rd(gen), rd(gen));
+            p.getVector3fMap() += baseOffset.getVector3fMap();
+            return p;
+          };
+          auto outsidePointGenerator = [this, &baseOffset] () {
+            pcl::PointXYZ p(rd(gen) + 2., rd(gen) + 2., rd(gen) + 2.);
+            p.getVector3fMap() += baseOffset.getVector3fMap();
+            return p;
+          };
+          pcl::CropHull<pcl::PointXYZ> cropHullFilter = createDefaultCropHull(inputCloud);
+          cropHullFilter.setDim(3);
+          std::vector<TestData> testDataSuite = createTestDataSuite(insidePointGenerator, outsidePointGenerator);
+          for (TestData const & testData : testDataSuite)
+          {
+            checker(cropHullFilter, testData);
+          }
+        }
+      }
+    private:
+      std::mt19937 gen;
+      std::uniform_real_distribution<float> rd;
+      pcl::PointCloud<pcl::PointXYZ>::Ptr baseInputCloud;
+  };
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_simple_test)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setInputCloud(testData.inputCloud);
+    pcl::Indices filteredIndices;
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+  };
+  test(checker);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_test_reentrancy)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setInputCloud(testData.inputCloud);
+    pcl::Indices filteredIndices;
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+    cropHullFilter.filter(filteredIndices);
```

Comment:
Seems to be duplicated check wrt 2 lines before

Reply:
this test check reentrancy of class. after second call of filter result must be the same as after first call.
PS: maybe I not correct use termin of reentrancy and must rename to repeatability

Reply:
Pure function?